test/test_prototype_transforms_functional.py

```diff
@@ -427,3 +461,74 @@ def test_correctness_affine_bounding_box_on_fixed_input(device):
     assert len(output_boxes) == len(expected_bboxes)
     for a_out_box, out_box in zip(expected_bboxes, output_boxes.cpu()):
         np.testing.assert_allclose(out_box.cpu().numpy(), a_out_box)
+
+
+@pytest.mark.parametrize("angle", [-54, 56])
+@pytest.mark.parametrize("translate", [-7, 8])
+@pytest.mark.parametrize("scale", [0.89, 1.12])
+@pytest.mark.parametrize("shear", [4])
+@pytest.mark.parametrize("center", [None, (12, 14)])
+def test_correctness_affine_segmentation_mask(angle, translate, scale, shear, center):
+    def _compute_expected_mask(mask, angle_, translate_, scale_, shear_, center_):
+        assert mask.ndim == 3 and mask.shape[0] == 1
+        affine_matrix = _compute_affine_matrix(angle_, translate_, scale_, shear_, center_)
+        inv_affine_matrix = np.linalg.inv(affine_matrix)
+        inv_affine_matrix = inv_affine_matrix[:2, :]
+
+        expected_mask = torch.zeros_like(mask.cpu())
+        for out_y in range(expected_mask.shape[1]):
+            for out_x in range(expected_mask.shape[2]):
+                output_pt = np.array([out_x + 0.5, out_y + 0.5, 1.0])
+                input_pt = np.floor(np.dot(inv_affine_matrix, output_pt)).astype(np.int32)
+                in_x, in_y = input_pt[:2]
+                if 0 <= in_x < mask.shape[2] and 0 <= in_y < mask.shape[1]:
+                    expected_mask[0, out_y, out_x] = mask[0, in_y, in_x]
+        return expected_mask.to(mask.device)
+
+    for mask in make_segmentation_masks():
+        output_mask = F.affine_segmentation_mask(
+            mask,
+            angle=angle,
+            translate=(translate, translate),
+            scale=scale,
+            shear=(shear, shear),
+            center=center,
+        )
+        if center is None:
+            center = [s // 2 for s in mask.shape[-2:][::-1]]
+
+        if mask.ndim < 4:
+            masks = [
+                mask,
+            ]
+        else:
+            masks = [m for m in mask]
+
+        expected_masks = []
+        for mask in masks:
+            expected_mask = _compute_expected_mask(mask, angle, (translate, translate), scale, (shear, shear), center)
+            expected_masks.append(expected_mask)
+        if len(expected_masks) > 1:
+            expected_masks = torch.stack(expected_masks)
+        else:
+            expected_masks = expected_masks[0]
+        torch.testing.assert_close(output_mask, expected_masks)
+
+
+@pytest.mark.parametrize("device", cpu_and_gpu())
+def test_correctness_affine_segmentation_mask_on_fixed_input(device):
+    # Check transformation against known expected output
+    # Rotate 90 degrees and scale
+    mask = torch.zeros(1, 32, 32, dtype=torch.long, device=device)
+    mask[0, 2:10, 2:10] = 1
+    mask[0, 32 - 9 : 32 - 3, 3:9] = 2
+    mask[0, 1:11, 32 - 11 : 32 - 1] = 3
+    mask[0, 16 - 4 : 16 + 4, 16 - 4 : 16 + 4] = 4
```

Comment:
This is a data creation. I'll leave a comment to make it more clear.

Reply:
The test checks F.affine_segmentation_mask with a degenerate configuration vs how it could be done using pure torch ops on cpu and cuda.
Input data does not play much role just it shouldn't be trivial as a single mask somewhere such that it would disappear after the transformation.

Reply:
@vfdev-5 could have another look into this? Unless I'm missing something in my comment above, I think we can simplify this test so it is easier to follow.

Reply:
In your suggestion with a single off-center rectangle we test the op on a binary but here I check on multi-class mask. It is a detail, but I prefer to ensure that multi-class mask is transformed correctly.

Reply:
Isn't a multi-class mask already tested by the reference test? In my understanding this test is a "smoke test" with a simple, easily verifiable example. What did you have in mind for this test?

Reply:
The goal for this test to ensure correctness on a "simple" case and check cpu/cuda devices as adding cuda to `test_correctness_affine_segmentation_mask` would make CI time too large.

Reply:
Got it, I missed that. Could you add this as a comment to the test? Maybe we can still reduce the number of masks to 2?